lib/presentation/view/context_menu/context_menu_action_builder.dart

```diff
@@ -0,0 +1,67 @@
+// suite, developed by Linagora.
+//
+// Copyright (C) 2020 LINAGORA
+//
+// This program is free software: you can redistribute it and/or modify it under the
+// terms of the GNU Affero General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later version,
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf
+//
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses
+// for the GNU Affero General Public License version
+//
+// 3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf
+// for
+//
+// the Additional Terms applicable to LinShare software.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/color_extension.dart';
+import 'package:meta/meta.dart';
+typedef OnContextMenuActionClick<T> = void Function(T data);
+
+abstract class ContextMenuActionBuilder<T> {
+  @protected final Key key;
+  @protected final SvgPicture actionIcon;
+  @protected final String actionName;
+  @protected OnContextMenuActionClick<T> onContextMenuActionClick;
+
+  ContextMenuActionBuilder(
+      this.key,
+      this.actionIcon,
+      this.actionName);
+
+  ContextMenuActionBuilder onActionClick(OnContextMenuActionClick<T> onContextMenuActionClick) {
+    this.onContextMenuActionClick = onContextMenuActionClick;
+    return this;
+  }
+
+  TextStyle actionTextStyle() {
+    return TextStyle(
+        fontSize: 16,
+        color: AppColor.documentNameItemTextColor);
```

Comment:
Since we're using this color a lot of times, we should rename it to a more generic name wdyt?

Reply:
I think action color in context menu is similar to documentName color. But for sure, it is better if we have more generic color. But at this time, I don't want to worst more time on that PR.